src/python_testing/matter_testing_support.py

```diff
@@ -878,6 +879,45 @@ def get_test_desc(self, test: str) -> str:
         except AttributeError:
             return test
 
+    # Sends an out-of-band command to a Matter app
+    def write_to_app_pipe(self, command):
+        """
+        Use the following environment variables:
+
+         - LINUX_DUT_IP 
+            * if not provided, the Matter app is assumed to run on the same machine as the test,
+              such as during CI, and the commands are sent to it using a local named pipe
+            * if provided, the commands for writing to the named pipe are forwarded to the DUT
+        - LINUX_DUT_UNAME
+            * if LINUX_DUT_IP is provided, use this for the DUT user name
+            * if LINUX_DUT_NAME is not provided, a default value is used ("root")
+            * If a remote password is needed, set up ssh keys to ensure that this script can log in to the DUT without a password:
+                 + Step 1: If you do not have a key, create one using ssh-keygen
+                 + Step 2: Authorize this key on the remote host: run ssh-copy-id user@ip once, using your password
+                 + Step 3: From now on ssh user@ip will no longer ask for your password
+        """
+        import os
+        dut_ip = os.getenv('LINUX_DUT_IP')
+
+        if dut_ip is None:
+            with open(self.app_pipe, "w") as app_pipe:
```

Comment:
self.app_pipe is not part of the base class, so this will be a problem if called without knowing `self.app_pipe` needed to be set. It also requires duplicating the PID extraction code... Please use a path prefix argument in `write_to_app_pipe`, and please append the PID to the path from the APP PID.

Reply:
I have added a comment below suggesting that we should be adding `app_pipe` to the base class since this is only set once at the start. It would make the calls of this method cleaner.

Or your comment
> will be a problem if called without knowing `self.app_pipe` needed to be set.
This is explained in the doc. Admittedly this is not explained it the method's doc so it should be there as well.

> It also requires duplicating the PID extraction code
If we remove this, when someone runs the app manually, there wouldn't be a way for the user to pass a PID, no? And in that case, it cannot be detected.